lib/codegen/meta/src/isa/mod.rs

```diff
@@ -0,0 +1,399 @@
+use cranelift_entity::PrimaryMap;
+use cranelift_entity::EntityRef;
+mod x86;
+mod arm32;
+//mod arm64;
+//mod riscv;
+
+fn is_power_of_two(n: u8) -> bool {
+    return n > 0 && n & (n - 1) == 0;
+}
+
+fn next_power_of_two(x: u8) -> u8 {
+    1 << (8 - x.leading_zeros())
+}
+
+/// Represents known ISA target.
+#[derive(Copy, Clone)]
+pub enum Isa {
+    Riscv,
+    X86,
+    Arm32,
+    Arm64,
+}
+
+impl Isa {
+    /// Creates isa target using name.
+    pub fn new(name: &str) -> Option<Self> {
+        Isa::all()
+            .iter()
+            .cloned()
+            .filter(|isa| isa.name() == name)
+            .next()
+    }
+
+    /// Creates isa target from arch.
+    pub fn from_arch(arch: &str) -> Option<Isa> {
+        Isa::all()
+            .iter()
+            .cloned()
+            .filter(|isa| isa.is_arch_applicable(arch))
+            .next()
+    }
+
+    /// Returns all supported isa targets.
+    pub fn all() -> [Isa; 4] {
+        [Isa::Riscv, Isa::X86, Isa::Arm32, Isa::Arm64]
+    }
+
+    /// Returns name of the isa target.
+    pub fn name(&self) -> &'static str {
+        match *self {
+            Isa::Riscv => "riscv",
+            Isa::X86 => "x86",
+            Isa::Arm32 => "arm32",
+            Isa::Arm64 => "arm64",
+        }
+    }
```

Comment:
I know this is pre-existing code, but could you move this code to a `Display` implementation while here? I just converted https://github.com/CraneStation/cranelift/pull/581.